src/components/Layout/ApiReferenceMenu.tsx

```diff
@@ -1,217 +1,190 @@
-            ocApi={ocApi}
-            sectionChange={sectionChange}
-            resourceChange={resourceChange}
-            operationChange={operationChange}
-            activeIndex={activeIndex}
-            selectedOperation={selectedOperation}
-          />
-        )
-      })}
-    </React.Fragment>
+    <aside>
+      {data.map(s => (
+        <ApiReferenceMenuSection key={s.path} data={s} uri={uri} />
+      ))}
+    </aside>
   )
 }
 
-interface SectionProps {
-  sectionChange: (sectionIndex: number) => void
-  resourceChange: (resourceName: string) => void
-  operationChange: (operation: ApiOperation) => void
-  section: ApiSection
-  ocApi: OrderCloudProps
-  activeIndex: number
-  selectedOperation: ApiOperation
-}
-function Section(props: SectionProps) {
-  const {
-    section,
-    ocApi,
-    sectionChange,
-    activeIndex,
-    resourceChange,
-    operationChange,
-    selectedOperation,
-  } = props
-
-  const classes = useStyles(props)
-  const sectionIndex = _findIndex(
-    ocApi.sections,
-    sect => sect['x-id'] === section['x-id']
-  )
-  const isActive = sectionIndex === activeIndex
-  const [open, setOpen] = React.useState(isActive)
-
-  const resources = ocApi.resources.filter(
-    r => r['x-section-id'] == section['x-id']
-  )
-
-  function handleClick() {
-    setOpen(!open)
-    sectionChange(sectionIndex)
-  }
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    linkItem: (props: any) => {
+      const defaultColor = theme.palette.getContrastText(
+        theme.palette.background.paper
+      )
+      const typeStyles = (() => {
+        switch (props.type) {
+          case 'section':
+            return {
+              color: props.isActive
+                ? theme.palette.secondary.main
+                : defaultColor,
+              marginBottom: theme.spacing(1),
+              maxWidth: '80%',
+            }
+          case 'resource':
+            return {
+              color: defaultColor,
+            }
+          case 'operation':
+            return {
+              color: props.isActive ? defaultColor : theme.palette.grey[500],
+              marginBottom: theme.spacing(0.5),
+              ...(props.isActive
+                ? {
+                    marginLeft: -theme.spacing(2.25),
+                    borderLeft: `${theme.spacing(0.5)}px solid ${
+                      theme.palette.secondary.light
+                    }`,
+                    paddingLeft: theme.spacing(1.75),
+                  }
+                : {}),
+            }
+          default:
+            return {}
+        }
+      })()
+      return {
+        textDecoration: 'none',
+        fontWeight: props.isActive ? 'bold' : undefined,
+        ...typeStyles,
+      }
+    },
+  })
+)
 
+const ApiReferenceMenuSection = (props: {
+  data: ApiReferenceMenuSection
+  uri: string
+}) => {
+  const { data, uri } = props
+  const isActive = useMemo(() => {
+    return uri.includes(data.path)
+  }, [uri])
```

Comment:
I did a little research on both `useMemo` and `useCallback` and found [this](https://kentcdodds.com/blog/usememo-and-usecallback) little nugget. 

You don't have to change anything on this PR but I think going forward let's keep usage of those for either computationally expensive calculations or scenarios where referential equality would cause it to re-evaluate every time. If performance isn't really a factor let's prefer readability and simplicity